File: applets/src/org/rr0/ufoathome/view/draw/CircleShape.java
package org.rr0.ufoathome.view.draw;

import java.awt.*;

/**
 * @author Jerôme Beau
 * @version 22 sept. 2003 21:49:45
 */
public class CircleShape extends OvalShape {
    public CircleShape(Component component) {
        super(component);
    }

    public void setWidth(int width) {
        super.setWidth(width);
        super.setHeight(width);
    }

    public void setHeight(int height) {
        super.setHeight(height);
        super.setWidth(height);
    }

}

File: applets/src/org/rr0/ufoathome/view/draw/CircleShape.ts
import {Component} from "../gui/Component";
import {Color} from "../gui/Color";
import {OvalShape} from "./OvalShape";

export class CircleShape extends OvalShape {
  constructor(component: Component) {
    super(component);
  }

  public setWidth(width: number) {
    super.setWidth(width);
    super.setHeight(width);
  }

  public setHeight(height: number) {
    super.setHeight(height);
    super.setWidth(height);
  }

  setCenterX(number: number) {

  }

  setCenterY(number: number) {

  }

  setColor(color: Color) {

  }
}
